node/readline/promises.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { Readline } from "../internal/readline/promises.mjs";
+
+import {
+  Interface as _Interface,
+  kQuestion,
+  kQuestionCancel,
+} from "../internal/readline/interface.mjs";
+import { AbortError } from "../internal/errors.ts";
+import { validateAbortSignal } from "../internal/validators.mjs";
+
+import { kEmptyObject } from "../internal/util.mjs";
+import type { Abortable } from "../_events.d.ts";
+
+// NOTE: These type redefinitions are intended to avoid circular dependencies.
+// import type {
+//   AsyncCompleter,
+//   Completer,
+//   ReadLineOptions,
+// } from "../_readline.d.ts";
```

Comment:
Instead of duplicating things, can they be placed in a file that both implementations can import?

Reply:
Thank you for your suggestion.
I have created another .d.ts file and modified them to import from the file. (caa1377)